specs/Separate_conf_files.adoc

```diff
@@ -0,0 +1,141 @@
+= Separate out per-component configuration files
+
+Configuration details for the tendrl modules are read from same file(tendrl.conf).
+Based on the tendrl modules we have to keep the configuration details into different
+files with proper name and read it independently.
+
+== Problem description
+
+Currently all the configuration details for different modules are read from same file .
+We have to split the configuration details as per modules and keep that into different
+file with proper name. Each configuration file should have its own and relevant configuration
+information only.
+
+We have to modify the python file (config.py) which read configuration file as per this
+modification. The python file should read each configuration files and it has to combine all
+the information into single object.
+
+== Use Cases
+
+None
+
+== Proposed change
+
+** Modify the sample configuration files in each modules. It should contains its own
+   and relevant configuration information only.
+** Rename the sample configuration files in each modules with proper name.
+** In common repository modify spec file to create a new directory inside /etc/tendrl/ in
+   the name of conf.d.
+** Modify the spec file in each modules to copy the sample configuration file into proper
+   directory:
+      * From common repository modify the spec file to copy the sample configuration file
+        inside the directory /etc/tendrl.
+      * From gluster_integration, ceph_integration and node_agent modify the each spec files
+        to copy the sample configuration files inside the directory /etc/tendrl/conf.d.
+** Modify config.py from common repository to read each configuration files independently
+   and it should combine all the information into single object.
+** From node_agent repository modify copy statement in /doc/source/installation.rst
+   based on new configuration file path change.
+** From gluster_integration repository modify copy statement in /doc/source/installation.rst
+   based on new configuration file path change.
+** From ceph_integration repository modify copy statement /doc/source/installation.rst based
+   on new configuration file path change.
+
+== Alternatives
+
+None
+
+== Data model impact
+
+None
+
+== Impacted modules
+
+=== Tendrl/common impact
+
+** /doc/source/installation.rst
+** /etc/tendrl/tendrl.conf.sample
+** config.py
+** tendrl-common.spec
+
+=== Tendrl/node_agent impact
+
+** /doc/source/installation.rst
+** etc/tendrl/tendrl.conf.sample
+** config.py
+** tendrl-node-agent.spec
+
+=== Tendrl/gluster_integration impact
+
+** /doc/source/installation.rst
+** etc/tendrl/tendrl.conf.sample
+** config.py
+** tendrl-gluster-integration.spec
+
+=== Tendrl/ceph_integration impact
+
+** /doc/source/installation.rst
+** etc/tendrl/tendrl.conf.sample
+** config.py
+** tendrl-ceph-integration.spec
+
+
+== Security impact
+
+None
+
+== Notifications/Monitoring impact
```

Comment:
Even notifications and monitoring modules would contain config files and should be taken care